magicctapipe/scripts/lst1_magic/lst1_magic_mc_dl0_to_dl1.py

```diff
@@ -83,28 +79,24 @@ def mc_dl0_to_dl1(input_file, output_dir, config):
         Configuration for the LST-1 + MAGIC analysis
     """
 
-    assigned_tel_ids = config["mc_tel_ids"]
+    assigned_tel_ids = config["mc_tel_ids"] #This variable becomes the dictionary {'LST-1': 1, 'MAGIC-I': 2, 'MAGIC-II': 3}
 
-    logger.info("\nAssigned telescope IDs:")
+    logger.info("\nAssigned telescope IDs:")    #Here we are just adding infos to the log file
     logger.info(format_object(assigned_tel_ids))
 
-    tel_id_lst1 = assigned_tel_ids["LST-1"]
-    tel_id_m1 = assigned_tel_ids["MAGIC-I"]
-    tel_id_m2 = assigned_tel_ids["MAGIC-II"]
-
     # Load the input file
     logger.info(f"\nInput file: {input_file}")
 
     event_source = EventSource(
         input_file,
-        allowed_tels=list(assigned_tel_ids.values()),
-        focal_length_choice="effective",
+        allowed_tels=list(filter(lambda check_id: check_id > 0, assigned_tel_ids.values())),  #Here we load the events for all telescopes with ID > 0.
+        focal_length_choice=focal_length,
```

Comment:
In some MC files (e.g., the old ones by Yoshiki we used to test this pipeline last year and to 'study' the PMTs performance before the new 4-LSTs production) the effective focal length is not saved. Maybe there are some other MC productions where only the nominal focal length can be found, so this is needed to analyze them. Moreover, this way you can compare two different analyses, nominal and effective focal length, to check for biases, to compare with MARS or to study coma aberration. In ctapipe 0.12 I think the default is nominal, in ctapipe 0.19 effective, so we cannot rely on default values. So (one of the) reason(s) is: improved flexibility (on MCs; be careful if you analyze real data too because in ctapipe_io_magic the effective one is the default one)

Reply:
I do not think we will use the nominal one, and the MCs that only had this one are long outdated, but having the flexibility is not bad, and the default is "effective" so fine to keep it like this.